File: src/main/java/xyz/hiziki/config/ConfigFile.java
package xyz.hiziki.config;

import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.plugin.java.JavaPlugin;
import xyz.hiziki.Main;

public class ConfigFile
{
    private final JavaPlugin plugin = Main.getPlugin();

    private final FileConfiguration config = plugin.getConfig();

    public boolean getENABLE_SET_HOME_SOUND() { return config.getBoolean("enable-set-home-sound"); }

    public boolean getENABLE_SET_HOME_MESSAGE() { return config.getBoolean("enable-set-home-message"); }

    public String getSET_HOME_MESSAGE() { return config.getString("set-home-message"); }

    public boolean getENABLE_SET_HOME_DELAY() { return config.getBoolean("enable-set-home-delay"); }

    public int getSET_HOME_DELAY() { return config.getInt("set-home-delay(s)"); }

    public boolean getENABLE_TELEPORT_SOUND() { return config.getBoolean("enable-teleport-sound"); }

    public boolean getENABLE_TELEPORT_MESSAGE() { return config.getBoolean("enable-teleport-message"); }

    public String getTELEPORT_MESSAGE() { return config.getString("teleport-message"); }

    public boolean getENABLE_TELEPORT_DELAY() { return config.getBoolean("enable-teleport-delay"); }

    public int getTELEPORT_DELAY() { return config.getInt("teleport-delay(s)"); }

    public int getMAX_HOME() { return config.getInt("max-home"); }

    //# --- 設定ファイル --- #
    //
    //# --- ホームを設定した時　--- #
    //enable-set-home-sound: true
    //enable-set-home-message: true
    //set-home-message: 'ホームを設定しました。'
    //enable-set-home-delay: false
    //set-home-delay(s): 5
    //
    //# --- ホームにTPした時 ---#
    //enable-teleport-sound: true
    //enable-teleport-message: true
    //teleport-message: 'ホームにTPしました。'
    //enable-teleport-delay: true
    //teleport-delay(s): 5
    //
    //# --- 最大ホーム数 --- #
    //max-home: 3
}

File: src/main/java/xyz/hiziki/config/ConfigFile.kt
package xyz.hiziki.config

import org.bukkit.plugin.java.JavaPlugin
import xyz.hiziki.Main

class ConfigFile
{
    private val plugin : JavaPlugin = Main.plugin!!

    private val config = plugin.config

    val enableSetHomeSound = config.getBoolean("enable-set-home-sound")

    val enableSetHomeDelay = config.getBoolean("enable-set-home-delay")

    val setHomeDelay = config.getInt("set-home-delay(s)")

    val enableTeleportSound = config.getBoolean("enable-teleport-sound")

    val enableTeleportDelay = config.getBoolean("enable-teleport-delay")

    val teleportDelay = config.getInt("teleport-delay(s)")

    val language = config.getString("language(ja, en)")

    val maxHome = config.getInt("max-home")

    val moveCancel = config.getBoolean("move-cancel")


    //# --- configuration --- #
    //
    //# --- set to the home　--- #
    //enable-set-home-sound: true
    //enable-set-home-delay: false
    //set-home-delay(s): 5
    //
    //# --- teleport to the home ---#
    //enable-teleport-sound: true
    //enable-teleport-delay: true
    //teleport-delay(s): 5
    //
    //# --- generic setting --- #
    //language(ja, en): en
    //max-home: 3
    //move-cancel: true
}